packages/flutter_tools/lib/src/custom_devices/custom_device_config.dart

```diff
@@ -161,4 +522,60 @@ class CustomDeviceConfig {
       screenshotCommand: explicitScreenshotCommand ? screenshotCommand : (screenshotCommand ?? this.screenshotCommand),
     );
   }
+
+  @override
+  bool operator ==(Object other) {
+    return other is CustomDeviceConfig
+      && other.id == id
+      && other.label == label
+      && other.sdkNameAndVersion == sdkNameAndVersion
+      && other.platform == platform
+      && other.enabled == enabled
+      && _listsEqual(other.pingCommand, pingCommand)
+      && _regexesEqual(other.pingSuccessRegex, pingSuccessRegex)
+      && _listsEqual(other.postBuildCommand, postBuildCommand)
+      && _listsEqual(other.installCommand, installCommand)
+      && _listsEqual(other.uninstallCommand, uninstallCommand)
+      && _listsEqual(other.runDebugCommand, runDebugCommand)
+      && _listsEqual(other.forwardPortCommand, forwardPortCommand)
+      && _regexesEqual(other.forwardPortSuccessRegex, forwardPortSuccessRegex)
+      && _listsEqual(other.screenshotCommand, screenshotCommand);
+  }
+
+  @override
+  int get hashCode {
```

Comment:
I think we might already have a quiver dep for hashcode you should consider instead

Reply:
you're right, using that instead

Reply:
btw I couldn't use this ([`hashObjects`](https://pub.dev/documentation/quiver/latest/quiver.core/hashObjects.html)) because the quiver flutter uses is still not null-migrated, so I can't use it inside `custom_device_config.dart`